src/main/drivers/barometer/barometer_lps22df.c

```diff
@@ -282,11 +282,21 @@ bool lps22dfDetect(baroDev_t *baro)
     // Reset the device
     busWriteRegister(dev, LPS22DF_CTRL_REG2, LPS22DF_CTRL_REG2_SWRESET);
 
-    // Enable one-shot ODR and averaging at 16
-    busWriteRegister(dev, LPS22DF_CTRL_REG1, LSM6DSV_ENCODE_BITS(LPS22DF_CTRL_REG1_ODR_ONE_SHOT,
+    // Wait for the device to be ready
+    uint8_t lps22df_chip_swreset = 0;
+    busReadRegisterBuffer(dev, LPS22DF_CTRL_REG2, &lps22df_chip_swreset, 1);
+    while (lps22df_chip_swreset & LPS22DF_CTRL_REG2_SWRESET) {}
```

Comment:
(or simpler variant of AI suggestion)
```suggestion
    int timeout = 100;  // bail out eventually, i2c timing will set delay
    do { 
        busReadRegisterBuffer(dev, LPS22DF_CTRL_REG2, &lps22df_chip_swreset, 1);
    } while (lps22df_chip_swreset & LPS22DF_CTRL_REG2_SWRESET || --timeout);
```

Reply:
I've done something similar